File: examples/javawebdriver/ChromeExample.java
package javawebdriver;

import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import io.github.bonigarcia.wdm.WebDriverManager;

public class ChromeExample {

	public static void main(String[] args) {
		WebDriverManager.chromedriver().setup();
		WebDriver driver = null;
		try {
			Path currentRelativePath = Paths.get("chrome-modheader/modheader.crx");
			ChromeOptions options = new ChromeOptions();
			options.addExtensions(new File(currentRelativePath.toAbsolutePath().toString()));
			driver = new ChromeDriver(options);
			driver.get("https://webdriver.modheader.com/add?test=ModHeader%20Test");

			driver.get("https://modheader.com/headers");
			new WebDriverWait(driver, Duration.ofSeconds(1))
					.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("body"), "ModHeader Test"));
			System.out.println("Success");
		} finally {
			if (driver != null) {
				driver.quit();
			}
		}
	}

}

File: examples/python/chrome-example.py
import os
from selenium import webdriver
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager

dirname = os.path.dirname(__file__)
extension_path = os.path.abspath(os.path.join(dirname, '../../chrome-modheader/modheader.crx'))
options = webdriver.ChromeOptions()
options.add_extension(extension_path)
driver = webdriver.Chrome(options=options, service=Service(ChromeDriverManager().install()))
driver.get("https://webdriver.modheader.com/add?test=ModHeader%20Test")
WebDriverWait(driver, 1).until(EC.title_is("ModHeader Done"))

driver.get("https://modheader.com/headers")
WebDriverWait(driver, 1).until(EC.text_to_be_present_in_element((By.TAG_NAME, "body"), "ModHeader Test"))
print('Success')

